parts/dcosagentcustomscript.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -e
+
+update_attribute()
+{
+    if [ ! -a /etc/mesosphere/roles/master ];
+    then
+        echo adding FD / UD attributes
+        meta=$( curl http://169.254.169.254/metadata/v1/InstanceInfo )
+        # parsing with bash to avoid jq install. This might break when the metadata service changes
+        ud="UD"$( echo $meta | cut -d\" -f 8)
+        fd="FD"$( echo $meta | cut -d\" -f 12)
+
+        if [ -a /var/lib/dcos/mesos-slave-common ];
+        then
+            echo Adding $ud $fd
+            sed -e "s/;*$/;UD:"$ud";FD:"$fd"/" -i /var/lib/dcos/mesos-slave-common
+        else
+            echo new attributes
+            # mkdir just in case. Didnt seem to exist reliably when the script is running

```

Comment:
to ensure you avoid what Jack is talking about, you can add "mkdir -p  /var/lib/dcos" in the runcmd sections of each of the dcos custom scripts